app/models/pieces.py

```diff
@@ -28,7 +27,26 @@ class Bishop(Piece): ...
 class Rook(Piece): ...
 
 
-class Queen(Piece): ...
+class Queen(Piece):
+    def validate_move(self, board: np.array, move: PieceMove):
```

Comment:
It looks good!
However we still have the scenario where the "from" and "to" position might be the same, where we should raise an error.
Since this is a check regardless the piece, we can add it under the "Piece.validate_move()" to be inherited by all piece classes.

Reply:
resolved as well